low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -252,13 +301,25 @@ def connect_to_prop_changed_signal(self, service):
             logger.debug(f"{service}({pid}) state is {state}:{substate}")
 
             if state in  ["activating", "reloading", "deactivating"]:
-                self.not_active_services[service] = \
-                                    [self.current_time(), "N/A", "N/A"]
-            elif state != "active":
+                if service not in self.not_active_services:
+                    self.not_active_services[service] = \
+                                    [self.current_time(), state, substate]
+            elif state != "active" and service not in self.failed_services:
                 self.failed_services.append(service)
-                self.raise_alert(service, "N/A", state, "N/A", substate,
-                                    "N/A", pid, 0)
+                self.raise_alert(service, prev_state, state, prev_substate,
+                                 substate, prev_pid, pid, 0)
                 logger.error(f"{service} is not active initially. state = {state}:{substate}")
+            elif state == "active":
+                if service in self.failed_services:
+                    self.failed_services.remove(service)
```

Comment:
@sumedhak27 If you remove service from list just before updating cache, then also it adds wrong data to cache?

Reply:
no if I remove the service before updating cache, then it's working all good.

Reply:
updated the flow.